### PR TITLE
Substitute secp256k1 library for the L1 version

### LINKED ISSUE #21: Remove secp256k1 package
This package is causing linker errors in the sequencer repo, due to duplicate C symbols for the crypto functions.

### PR DESCRIPTION
**Pull request by jules**
Fixes #21

### CODE REVIEW

**Review by 0xmountaintop — APPROVED** [review]
LGTM!

**Comment by 0xmountaintop** [comment]
BTW, maybe also a good idea to add comments in the codes to explain why we would like to keep the ethereum's version

**Comment by 0xmountaintop** [comment]
after another look I think we should use go-ethereum v1.10.13, to be consistent with the statement in our readme

File: go.sum [hunk 1]
```diff
@@ -132,6 +132,8 @@ github.com/edsrzf/mmap-go v1.0.0 h1:CEBF7HpRnUCSJgGUb5h1Gm7e3VkmVDrR8lvWVLtrOFw=
 github.com/edsrzf/mmap-go v1.0.0/go.mod h1:YO35OhQPt3KJa3ryjFM5Bs14WD66h8eGKpfaBNrHW5M=
 github.com/envoyproxy/go-control-plane v0.9.1-0.20191026205805-5f8ba28d4473/go.mod h1:YTl/9mNaCwkRvm6d1a2C3ymFceY/DCBVvsKhRF0iEA4=
 github.com/envoyproxy/protoc-gen-validate v0.1.0/go.mod h1:iSmxcyjqTsJpI2R4NaDN7+kN2VEUnK/pcBlmesArF7c=
+github.com/ethereum/go-ethereum v1.10.15 h1:E9o0kMbD8HXhp7g6UwIwntY05WTDheCGziMhegcBsQw=
+github.com/ethereum/go-ethereum v1.10.15/go.mod h1:W3yfrFyL9C1pHcwY5hmRHVDaorTiQxhYBkKyu5mEDHw=
```
Comment (mask-pp) [inline go.sum]: We'd better stay the same version v1.10.13 !

Comment (jules) [reply go.sum]: Just changed it, should be on `1.10.13` now :)

**Review by mask-pp — APPROVED** [review]

**Pull request merged**